nixos/modules/programs/command-not-found/default.nix

```diff
@@ -0,0 +1,18 @@
+{ stdenv, rustPlatform, pkg-config, sqlite
+, dbPath ? "/nix/var/nix/profiles/per-user/root/channels/nixos/programs.sqlite" }:
+
+rustPlatform.buildRustPackage {
+  name = "command-not-found";
+  src = ./rust;
+
+  DB_PATH = dbPath;
+  NIX_SYSTEM = stdenv.system;
+
+  postInstall = ''
+    strip $out/bin/command-not-found
+  '';
+
+  buildInputs = [ sqlite ];
+  nativeBuildInputs = [ pkg-config ];
+  cargoSha256 = "13q61bb4b1q40g424pbssyp3ln79q1a33vmyz9s9wlqnac34cibd";
```

Comment:
We should really avoid the use of `cargoSha256` since it's basically an abuse of fixed-output derivations. See https://github.com/NixOS/nix/issues/2270.

Reply:
And it's not needed anymore for building Rust, since we can handle `Cargo.toml` using `builtins.fromTOML`. See for instance https://github.com/edolstra/import-cargo/blob/master/flake.nix.

Reply:
We can use it here but not in general, because it would mean that have to do import from derivation to load `Cargo.lock`

Reply:
Also this function is not part of nixpkgs and uses flake feature, which we cannot use yet. As this tool becomes obsolete once we switch from channel to flakes, we can ignore this as well.

Reply:
IMHO this is a general of `buildRustPackage` and not of this package/pr, so I am marking this as solved.

Reply:
Is it worth opening a tracking issue for improving `buildRustPackage`?

Reply:
Yes. We have some other candidates for it like `buildRustCrate` with https://github.com/kolloch/crate2nix